library/Vanilla/UploadedFile.php

```diff
@@ -196,7 +196,7 @@ public function generatePersistedUploadPath(string $persistSubDirectory = '', st
         $ext = strtolower(pathinfo($this->getClientFilename(), PATHINFO_EXTENSION));
         $baseName = basename($this->getClientFilename(), ".${ext}");
         $baseName = sprintf($nameFormat, $baseName);
-        $baseName = \Gdn_Format::url($baseName);
+        $baseName = \Gdn_Format::url(urlencode($baseName));
```

Comment:
I think I see that this is just working by coincidence, but will ultimately create URLs that are garbled in many cases. Consider a file with the name `"cät"`. That should give us a file with the name `"cat"`. But your changes will give us something like `"c-c3-a4t"`. Can we not remove the call to `Gdn_Format::url()` and just go with `rawurlencode()`?